lib/rubygems/exceptions.rb

```diff
@@ -20,6 +20,53 @@ class Gem::Exception < RuntimeError
 
 class Gem::CommandLineError < Gem::Exception; end
 
+class Gem::UnknownCommandError < Gem::Exception
+
+  class SpellChecker
```

Comment:
I think the SpellChecker class should live in its own file.

Reply:
Happy to move this somewhere else if you really think it needs to. For the most recent refactor though it was significantly easier to change since it was in the same file. If you feel strongly, where should it go?

Reply:
The class could be placed under `lib/rubygems`

Reply:
I've moved this into its own file.